### PR TITLE
Monitoring Backend

### PR DESCRIPTION
**Pull request by aginzberg**
Added two filters. One to grab all of a requesters modules (essentially identical to Elsa's requester_projects) and one to grab the data needed for monitoring (tasks for the data field, task_workers for the worker alias, and task_worker_results for the result).

### CODE REVIEW

File: crowdsourcing/viewsets/worker.py [hunk 1]
```diff
@@ -156,6 +155,16 @@ def retrieve(self, request, *args, **kwargs):
         serializer = TaskWorkerResultSerializer(instance=worker)
         return Response(serializer.data)
 
+    @list_route(methods=['POST'])
```
Comment (dmorina) [inline crowdsourcing/viewsets/worker.py]: you mean GET right?

File: static/js/monitor/services/monitor.service.js [hunk 2]
```diff
@@ -38,10 +38,13 @@
     }
 
 
-    function getTaskWorkerResults(moduleId){
+    function getMonitoringData(module_id){
       var settings = {
-        url: '/api/task/' + moduleId + '/',
-        method: 'GET'
+        url: '/api/task-worker-result/monitoring_data/',
+        method: 'POST',
```
Comment (dmorina) [inline static/js/monitor/services/monitor.service.js]: GET?

File: crowdsourcing/viewsets/worker.py [hunk 3]
```diff
@@ -156,6 +155,16 @@ def retrieve(self, request, *args, **kwargs):
...
+    def monitoring_data(self, request, **kwargs):
+        module = Module.objects.get(id=request.data['module_id'])
```
Comment (dmorina) [inline crowdsourcing/viewsets/worker.py]: test this after changing to GET, it will be empty, params are stored elsewhere

**Comment by elsabakiu** [comment]
@aginzberg cool, just keep in mind that project objects returned by the api(requester_projects) contain all the corresponding modules, the modules the corresponding tasks and so on. For example, you can get the modules as follows: modules = project.modules; (you can refer to the Monitor.getProject() function in monitor.controller.js for more examples) When possible, use this way to get the data rather then making a duplicate api call.

**Comment by dmorina** [comment]
@elsabakiu we will drop the tasks from `/api/project/` (just task count is sufficient) these serializers are dynamic so we can choose on what call which fields we need

**Comment by elsabakiu** [comment]
Yes, sounds even better. Send to the frontend only the data needed for a specific page + use related entities directly rather then making duplicate api calls :)

File: crowdsourcing/viewsets/worker.py [hunk 4]
```diff
@@ -156,6 +155,13 @@ def retrieve(self, request, *args, **kwargs):
         serializer = TaskWorkerResultSerializer(instance=worker)
         return Response(serializer.data)
 
+    @list_route(methods=['GET'])
+    def monitoring_data(self, request, **kwargs):
```
Comment (dmorina) [inline crowdsourcing/viewsets/worker.py]: this goes to task viewset

**Pull request closed without merging**